src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Services/Search/SearchService.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Azure;
+using Azure.Search.Documents;
+using Azure.Search.Documents.Models;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Services.Interfaces.Search;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Azure;
+
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Services.Search;
+
+public class SearchService(IAzureClientFactory<SearchClient> searchClientFactory, ILogger<SearchService> logger)
+    : ISearchService
+{
+    public async Task<Either<ActionResult, PaginatedListViewModel<PublicationSearchResult>>> SearchPublications(
+        int page,
+        int pageSize,
+        IEnumerable<Guid> publicationIds,
+        string? searchText = null,
+        CancellationToken cancellationToken = default
+    )
+    {
+        var searchClient = searchClientFactory.CreateClient("PublicationsSearchClient");
+        var options = new SearchOptions
+        {
+            Filter = CreatePublicationIdsFilter(publicationIds),
+            IncludeTotalCount = true,
+            Size = pageSize,
+            Skip = (page - 1) * pageSize,
+        };
+
+        try
+        {
+            SearchResults<PublicationSearchResult> response = await searchClient.SearchAsync<PublicationSearchResult>(
+                searchText,
+                options,
+                cancellationToken
+            );
+            var totalCount = Convert.ToInt32(
+                response.TotalCount
+                    ?? throw new InvalidOperationException(
+                        "Search response did not contain TotalCount even though IncludeTotalCount was requested."
+                    )
+            );
+            var allDocuments = await response
```

Comment:
Up to you but I find all these code blocks without any empty lines between quite dense to read.  Maybe it's just the GitHub page that makes it harder!

Reply:
Good point, I've added in some blank lines in-between now. Thanks. 🙂